### PR TITLE
[ROCM] fix softmax_with_cross_entropy_op, test=develop

### PR DESCRIPTION
**Pull request by ronny1996**
### PR types

Bug fixes
### PR changes

OPs
### Describe

fix softmax_with_cross_entropy_op

### CODE REVIEW

File: paddle/fluid/operators/softmax_with_cross_entropy_op.cu [hunk 1]
```diff
@@ -732,7 +727,7 @@ static void SoftmaxWithCrossEntropyFusedKernel(
 template <typename T>
 static void CrossEntropyFusedKernel(const T* logits_data, const T* labels_data,
                                     T* loss_data, int n, int d, int axis_dim,
-                                    cudaStream_t stream) {
+                                    gpuStream_t stream) {
```
Comment (chajchaj) [inline paddle/fluid/operators/softmax_with_cross_entropy_op.cu:730]: 这里为什么修改？

Comment (ronny1996) [reply paddle/fluid/operators/softmax_with_cross_entropy_op.cu:730]: https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/platform/type_defs.h
兼容cuda和rocm

**Review by chajchaj — APPROVED** [review]
LGTM

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**